File: pset6/credit.py

def main():
    while True:
        print("What's your credit card number")
        number = input()
        if len(number)>0:
            break

    single_digits_sum = 0;
    double_digits_sum = 0;

    for i in range(len(number)):
        if i%2 == 0:   # for those which are in single digtis, as i starts counting from 0.
            single_digits_sum += int(number[i])
        if i%2 !=0 :     # for those which are in double digits, as i starts counting from0.
            mult_2 = int(number[i])*2
            if mult_2 >= 10:
                double_digits_sum += (1+(mult_2-10))

            if mult_2 < 10:
                double_digits_sum += mult_2
    total_sum = single_digits_sum + double_digits_sum
    # print(total_sum)

    if total_sum % 10 == 0 and int(number[0]) == 3 and len(number) == 15:
        print("AMEX")
    elif total_sum % 10 == 0 and int(number[0]) == 5 and len(number) == 16:
        print("MASTERCARD")
    elif total_sum % 10 == 0 and int(number[0]) == 4:
        print("VISA")
    else:
        print("INVALID")

if __name__ == "__main__":
    main()

File: pset1/credit.c
#include <cs50.h>
#include <stdio.h>

long long number;
long long counter = 1;
int j = 0;
int every_each_sum1 = 0;
int every_each_sum2 = 0;

int main(void)
{
    do
    {
        printf("What's your credit card number?  ");
        number = get_long_long();
    }
    while (number<= 0);

    for(int i=0; i<16; i++)
    {
        long long every_other_digit = number / counter;
        counter = counter * 10;
        int molodul = every_other_digit % 10;
        // printf("%i", molodul);
        // if this is a odd digit.
        if (j%2 == 0)
        {
            every_each_sum1 = every_each_sum1 + molodul;
        }
        // if this is a even digit.
        else
        {
            int doubled_molodul = molodul * 2;
            if (doubled_molodul >= 10)
            {
                doubled_molodul = doubled_molodul%10 + 1;
            }
            every_each_sum2 = every_each_sum2 + doubled_molodul;
            // printf("%i\n", doubled_molodul);
        }
        j++;
    }
    int final_sum = every_each_sum1 + every_each_sum2;
    // printf("%i  ", every_each_sum1);
    // printf("%i  ", every_each_sum2);
    // printf("%i\n", final_sum);
    if (final_sum % 10 == 0 && number / 100000000000000 == 3)
    {
      printf("AMEX\n");
    }
    else if (final_sum % 10 == 0 && number / 1000000000000000 == 5)
    {
      printf("MASTERCARD\n");
    }
    else if (final_sum % 10 == 0 && number / 1000000000000000 == 4)
    {
      printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}